### DISCUSSION TITLE
cargo install kanidm_tools fail

### DISCUSSION THREAD

**Discussion by a137872798** [kanidm/kanidm#1809] [General]
when i install kanidm_tools with version  v1.1.0-alpha.12. 
tell me could not compile 'compact_jwt'.
if the rust version is not match?
what should i do

**Comment by yaleman** [comment]
What version of Rust are you using? The minimum Rust version is currently 1.66.

**Reply by a137872798** [reply]
1.70.0

**Reply by a137872798** [reply]
![image](https://github.com/kanidm/kanidm/assets/35627706/df77d143-44b8-40b1-91a5-e2229cbc608f)

**Reply by yaleman** [reply]
Ok, that actually like you're missing the dependencies listed in [the developer guide](https://kanidm.github.io/kanidm/stable/DEVELOPER_README.html#getting-started-for-developers). Please ensure you've got the OpenSSL libs installed because that'll be the error.

**Reply by a137872798** [reply]
oh，i see. My os is centos. Not in the list. I had solve the problem about "failed to run custom build command for openssl-sys "

**Reply by yaleman** [reply]
I just tested quickly in a Centos VM and the Fedora instructions seem to work to build all the packages:

```
sudo yum install -y rust cargo openssl-devel systemd-devel sqlite-devel pam-devel
```

**Reply by a137872798** [reply]
I run these instructions. But the question still exists.

**Reply by a137872798** [reply]
Maybe  my rust version is too new for compact_jwt 0.2.9?
The prompt  seem like grammer error.

**Reply by yaleman** [reply]
I tested on rust 1.70, it's not a rust version error. Why do you think it's the version?

**Reply by a137872798** [reply]
Thank you.You are right, it seems like condition compile fail.
![image](https://github.com/kanidm/kanidm/assets/35627706/48c34a4d-f392-4a11-a4c2-e7259f5b03dd)
![image](https://github.com/kanidm/kanidm/assets/35627706/c512b066-9719-4d34-a67d-deeedd60d800)

**Reply by a137872798** [reply]
But why after run these instructions (sudo yum install -y rust cargo openssl-devel systemd-devel sqlite-devel pam-devel),the question still exists.

**Reply by yaleman** [reply]
Did you clone the git repository and build from there? 

Is it the same error? Are you sure all the required packages installed?

Without more context it's not clear why your system is causing the build to fail.  This isn't a distribution we test on, and it looks like the error is not in one of our crates, but a dependency.

**Reply by Firstyear** [reply]
Centos is an extremely problematic distribution for us as they lag behind on many fronts.

Use the tools container instead, or use https://build.opensuse.org/repositories/network:idm

**Comment by a137872798** [comment]
Thanks for yours help, now i use tools container.
